### PR TITLE
Issue 1844 - hzn deploycheck on an edge cluster

### PR DESCRIPTION
**Pull request by sirotsinskuy**
Signed-off-by: Antonii Sirotsinskyi <asiro@softserveinc.com>

### CODE REVIEW

File: cli/deploycheck/policy.go [hunk 1]
```diff
@@ -60,6 +60,10 @@ func PolicyCompatible(org string, userPw string, nodeId string, nodeArch string,
 		policyCheckInput.NodePolicy = &np
 	}
 
+	if nodeType == "" {
```
Comment (linggao) [inline cli/deploycheck/policy.go:63]: nodeId may also be empty. So please use: 
if nodeType == "" && policyCheckInput.NodeId != ""

File: cli/hzn.go [hunk 2]
```diff
@@ -407,7 +407,7 @@ Environment Variables:
 	deploycheckLong := deploycheckCmd.Flag("long", msgPrinter.Sprintf("Show policies and userinput used for the compatibility checking.")).Short('l').Bool()
 	policyCompCmd := deploycheckCmd.Command("policy", msgPrinter.Sprintf("Check policy compatibility."))
 	policyCompNodeArch := policyCompCmd.Flag("arch", msgPrinter.Sprintf("The architecture of the node. It is required when -n is not specified. If omitted, the service of all the architectures referenced in the deployment policy will be checked for compatibility.")).Short('a').String()
-	policyCompNodeType := policyCompCmd.Flag("node-type", msgPrinter.Sprintf("The node type. The valid values are 'device' and 'cluster'. The default is 'device'.")).Short('t').Default("device").String()
+	policyCompNodeType := policyCompCmd.Flag("node-type", msgPrinter.Sprintf("The node type. The valid values are 'device' and 'cluster'. The default value is the type of node provided by -n (current registered device if nothing is provided).")).Short('t').String()
```
Comment (linggao) [inline cli/hzn.go:410]: Please change the last part of the text to:
"The default value is the type of the node provided by -n  or current registered device, if omitted."
Same for the other 2 places.

**Review by linggao — CHANGES_REQUESTED** [review]
per comment

File: cli/deploycheck/policy.go (hunk 1 above)
Comment (sirotsinskuy) [reply cli/deploycheck/policy.go]: Added nodeId check

File: cli/hzn.go (hunk 2 above)
Comment (sirotsinskuy) [reply cli/hzn.go]: Changed the description, thanks

**Review by linggao — DISMISSED** [review]

**Comment by linggao** [comment]
Please resolve the file conflict.

**Review by linggao — APPROVED** [review]

**Pull request merged**